gms/gossiper.cc

```diff
@@ -2388,7 +2388,7 @@ void gossiper::force_newer_generation() {
     eps.get_heart_beat_state().force_newer_generation_unsafe();
 }
```

Comment:
`storage_service: load_saved_endpoint of token owners with unknown state`

> However, a bootstrapping node, has no persistent knowledge of the down nodes,
> and it learns about their existance from gossip.
> But, since the down nodes have unknown status, they never go
> through `handle_state_normal`.

You mean any starting node (including restarts), not just bootstrapping?

Reply:
Wasn't the original idea to inject STATUS of peers? I'm confused. The PR or commit messages don't seem to explain the motivation behind this approach.

Reply:
> You mean any starting node (including restarts), not just bootstrapping?

No, restarting node that already joined the cluster will load the down-nodes state from their local system.peers table.

Reply:
> Wasn't the original idea to inject STATUS of peers? I'm confused. The PR or commit messages don't seem to explain the motivation behind this new approach.

That was the original idea, but per https://github.com/scylladb/scylladb/issues/15787#issuecomment-1774320071
this is not deemed safe according to @asias.